src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1720,20 +1720,48 @@ def update_load_balancer_profile(self, mc: ManagedCluster) -> ManagedCluster:
 
         :return: the ManagedCluster object
         """
+        existing_managed_outbound_ips = None
```

Comment:
BTW, we may open another PR to sort out the getters for these load balancer properties. I would vote for putting the logic of how/where to get the value into the getter functions. Besides, is it necessary to add cli checks to avoid setting the values of managed_outbound_ips, outbound_ips and outbound_ips_prefixes at the same time?

Reply:
> is it necessary to add cli checks to avoid setting the values of managed_outbound_ips, outbound_ips and outbound_ips_prefixes at the same time?

I think the challenging part about the cli is that the SDK will provide default values for some properties and not others. So the cli ends up checking many values against what the customer provided. AKS RP already handles default values so there really isn't a need for the SDK to set anything in my opinion. The only parameters required for RP are API version, subscription ID, and resource group. Everything else can be defaulted.

Reply:
But I agree with refactoring the getters for LB profile.